source/common/stats/statsd.cc

```diff
@@ -97,12 +97,55 @@ TcpStatsdSink::TlsSink::~TlsSink() {
   }
 }
 
+void TcpStatsdSink::TlsSink::beginFlush(bool expect_empty_buffer) {
+  ASSERT(!expect_empty_buffer || buffer_.length() == 0);
+  ASSERT(current_slice_mem_ == nullptr);
+  UNREFERENCED_PARAMETER(expect_empty_buffer);
+
+  uint64_t num_iovecs = buffer_.reserve(FlushSliceSizeBytes, &current_buffer_slice_, 1);
+  ASSERT(num_iovecs == 1);
+  UNREFERENCED_PARAMETER(num_iovecs);
+
+  ASSERT(current_buffer_slice_.len_ >= FlushSliceSizeBytes);
+  current_slice_mem_ = reinterpret_cast<char*>(current_buffer_slice_.mem_);
+}
+
+void TcpStatsdSink::TlsSink::commonFlush(const std::string& name, uint64_t value, char stat_type) {
+  ASSERT(current_slice_mem_ != nullptr);
+  // 40 > 6 (prefix) + 4 (random chars) + 30 for number (bigger than it will ever be)
+  if (current_buffer_slice_.len_ - usedBuffer() < name.size() + 40) {
```

Comment:
This math is scary. Can we have an `ASSERT` at the end of this method that ensures we haven't advanced further than we thought by this computation?